app/models/set_member_subject.rb

```diff
@@ -48,4 +54,10 @@ def remove_from_queues
   def set_random
     self.random = rand
   end
+
+  private
+
+  def timestamp_newly_retired_workflows
+    retired_subject_workflow_counts.each(&:retire!)
```

Comment:
Records just added to the collection (`retired_subject_workflow_counts << record` etc) will be there too.